### ISSUE TITLE
admin doesn't work against demo.api-platform.com

### ISSUE THREAD

**Issue by karser** [api-platform/admin#14]
***The exact error:***
```
Fetch API cannot load http://demo.api-platform.com/docs.jsonld. Response for preflight is invalid (redirect)
```
![image](https://cloud.githubusercontent.com/assets/1675033/26644007/00dbdd94-463c-11e7-8aa9-c68122256996.png)

***How to reproduce:***

1. I'm using https://demo.api-platform.com as an api endpoint
![image](https://cloud.githubusercontent.com/assets/1675033/26643835/5a39dda6-463b-11e7-8928-96dd7246da10.png)
OPTIONS response contains LINK header which refers to http://demo.api-platform.com/docs.jsonld (although I specified https).

2. It is requesting http://demo.api-platform.com/docs.jsonld and receives `301 Moved Permanently` and  `Location:https://demo.api-platform.com/docs.jsonld`
![image](https://cloud.githubusercontent.com/assets/1675033/26643945/b7e69642-463b-11e7-9192-c7204fcfed17.png)

Also notice that first response is HTTP/2 and second one is HTTP 1.1

**Comment by dunglas** [comment]
@karser thanks for the report.

@Simperfit do you think you can fix that? https://symfony.com/doc/current/security/force_https.html may help, in prod we should only generate `https` routes.

IMO it can be done by adding a security rule for the prod env in https://github.com/api-platform/demo/blob/master/app/config/security.yml

**Comment by wadjeroudi** [comment]
@dunglas I've the same problem. I think your solution using the security.yml will only deny if no https is used.
Maybe you should use https://symfony.com/doc/current/routing/scheme.html
But don't know how to do that only for prod env.

**Comment by dunglas** [comment]
@wadjeroudi good idea. We can define routes depending of the environment, so it should do the trick.

**Comment by mickaelandrieu** [comment]
@dunglas this issue is still here, I guess I can't help here as the fix you suggested needs to be done on your demonstration website, am I  right?

**Comment by dunglas** [comment]
Yes, @Simperfit is handling it (we'll publish soon a new demo)

**Comment by dunglas** [comment]
@mdewet is working on it on the demo deployment side.
